cpp/assetmanager.h

```diff
@@ -74,7 +79,7 @@ class AssetManager {
 	 * Map from inotify watch handle fd to texture instance ptr.
 	 * The kernel returns the handle fd when events are triggered.
 	 */
-	std::unordered_map<int, Texture *> watch_fds;
+	std::unordered_map<int, std::weak_ptr<Texture>> watch_fds;
```

Comment:
I would use a shared_ptr in this map, as well. The weak_ptr does not avoid any circular references and with every usage it is converted to a shared_ptr using the lock-function anyway.